### PR TITLE
Issue #7980: Resolve Pitest Issues - JavadocMethodCheck (8)

### LINKED ISSUE #7923: Resolve Pitest Issues - JavadocMethodCheck (2)
Child issue of https://github.com/checkstyle/checkstyle/issues/7797 ,

https://github.com/checkstyle/checkstyle/blob/f9131f9536c71d963c83aab1550b7a0371aa9ff5/.ci/pitest.sh#L183

### LINKED ISSUE #7924: Resolve Pitest Issues - JavadocMethodCheck (3)
Child issue of https://github.com/checkstyle/checkstyle/issues/7797 ,

https://github.com/checkstyle/checkstyle/blob/f9131f9536c71d963c83aab1550b7a0371aa9ff5/.ci/pitest.sh#L184

### LINKED ISSUE #7925: Resolve Pitest Issues - JavadocMethodCheck (4)
Child issue of https://github.com/checkstyle/checkstyle/issues/7797 ,

https://github.com/checkstyle/checkstyle/blob/f9131f9536c71d963c83aab1550b7a0371aa9ff5/.ci/pitest.sh#L185

### LINKED ISSUE #7980: Resolve Pitest Issues - JavadocMethodCheck (8)
Child issue of https://github.com/checkstyle/checkstyle/issues/7797 ,

https://github.com/checkstyle/checkstyle/blob/f9131f9536c71d963c83aab1550b7a0371aa9ff5/.ci/pitest.sh#L189

### PR DESCRIPTION
**Pull request by gaurabdg**
Resolves #7980 
Resolves #7923 
Resolves #7924 
Resolves #7925 
Pitest reports and UT explanation is available at
https://github.com/checkstyle/checkstyle/issues/7980#issuecomment-607199640
**Pitest report after UT addition**:
https://gaurabdg.github.io/checkstyle-tester-reports/pitest/JavadocMethodCheck-fixed/com.puppycrawl.tools.checkstyle.checks.javadoc/JavadocMethodCheck.java.html

The addition of this UT also resolves the survived mutations of #7923 #7924 #7925 
Diff Report: https://gaurabdg.github.io/checkstyle-tester-reports/regression/JavadocMethod/diff/

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java [hunk 1]
```diff
@@ -581,4 +584,32 @@ public void testWithoutLogErrors() throws Exception {
...
+                + "JavadocMethodCheck$Token");
+        final Constructor<?> tokenConstructor = tokenClazz.getDeclaredConstructor(String.class,
+                int.class, int.class);
+        final Object token = tokenConstructor.newInstance("B1", 21, 22);
+
+        final Method createClassInfo = clazz.getDeclaredMethod("createClassInfo", tokenClazz,
+                String.class);
+        createClassInfo.setAccessible(true);
+        final Object instanceResult = createClassInfo.invoke(obj, token,
+                "InputJavadocMethodClassAliasTest");
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java:609]: Concern at https://github.com/checkstyle/checkstyle/issues/7980#issuecomment-607237012

Comment (gaurabdg) [reply src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java:609]: @rnveach Replied in the issue comment thread.

**Comment by gaurabdg** [comment]
@rnveach Please reply.
The reasoning behind the implementation of this UT: https://github.com/checkstyle/checkstyle/issues/7980#issuecomment-607398059
Shall we keep this UT or remove the redundant code?

**Comment by strkkk** [comment]
@rnveach ping, please help to finish PR

@gaurabdg please rebase to latest master and resolve conflicts

File: src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java [hunk 2]
```diff
@@ -595,4 +598,32 @@ public void testWithoutLogErrors() throws Exception {
                 CommonUtil.EMPTY_STRING_ARRAY);
     }
 
+    @Test
+    public void testClassAliasResolving() throws Exception {
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheckTest.java:602]: I am still looking over all discussions but I wanted to point to https://github.com/checkstyle/checkstyle/issues/7980#issuecomment-607310043 . If reflection stays, we need javadoc on why reflection is needed.

**Comment by rnveach** [comment]
Waiting on confirmation on https://github.com/checkstyle/checkstyle/issues/7980#issuecomment-628220043 .

**Comment by gaurabdg** [comment]
Okay. So, I guess we should close this PR as @HuGanghui as already submitted a PR(#8113) with removed code which solves these issues.

**Comment by rnveach** [comment]
@gaurabdg It depends on what you want to do. It is unfortunate that these pitest coincide with other pitest issues. Our stance before has always been that the first to PR would finish it. We have been discussing your issue and PR and changes longer and no one has started reviewing his work.

**Comment by gaurabdg** [comment]
> @gaurabdg It depends on what you want to do if you want to continue or have the other PR take over. The other PR doesn't go far enough in everything that can be removed.
> 
> It is unfortunate that these pitest coincide with other pitest issues. Our stance on this problem has always been that the first to PR would finish it. We have been discussing your issue and PR and changes longer and no one has started reviewing his work and you notified him of the issue once you saw it. I did not as I have been too busy and have too big a back log.

Okay, I'll work on it.

**Comment by rnveach** [comment]
TC is failing with 3 errors.

**Comment by gaurabdg** [comment]
> TC is failing with 3 errors.

I have rebase on master, it will be resolved.
 
> Also we will need an updated regression.

What exactly needs to be changed? I didn't change the logic right?

**Comment by rnveach** [comment]
> I didn't change the logic right?

Technically you changed the code which code impact the logic, though we are pretty sure it will result in no change. Even I wasn't expecting all of `currentTypeParams` to be removed. I would rather be safe and just see an updated regression run to show check functionality is not impacted in a noticable way. Then I will approve the PR and we can get it merged.

**Comment by gaurabdg** [comment]
@rnveach Here's the regression report: https://gaurabdg.github.io/checkstyle-tester-reports/regression/JavadocMethod/diff/

**Pull request merged**